### PR TITLE
feat(gen9): Wave 7 -- held items + Booster Energy

### PR DESCRIPTION
**Pull request by uehlbran**
## Summary

- Add `Gen9Items.ts` with `applyGen9HeldItem` handler covering all five trigger points (`before-move`, `end-of-turn`, `on-damage-taken`, `on-contact`, `on-hit`)
- Wire `Gen9Ruleset.applyHeldItem` to delegate to the new handler
- Export 26 pure-function helpers from `packages/gen9/src/index.ts`
- 141 tests covering all item categories, edge cases, and Gen 9-specific mechanics

### Gen 9 differences from Gen 8
- **Fairy Feather** added to type-boost items (1.2x Fairy-type boost)
- **Booster Energy** helper for Protosynthesis/Quark Drive activation
- **Covert Cloak** helper for blocking secondary effects
- **Snow** replaces Hail for Icy Rock weather rock type
- No Dynamax suppression of Choice lock (Dynamax removed in Gen 9)
- No Z-Crystals or Mega Stones

### Item categories covered
Choice items, type-boost items (incl. plates, incenses, gems), type-resist berries, Life Orb, Leftovers, Black Sludge, Focus Sash/Band, Eviolite, Assault Vest, Rocky Helmet, Sitrus/Oran/Lum berries, status-cure berries, pinch stat berries, Weakness Policy, status orbs (Toxic/Flame), Booster Energy, Covert Cloak, Utility Umbrella, Safety Goggles, weather rocks, Shell Bell, King's Rock/Razor Fang, Throat Spray, Red Card, Eject Button, Mental Herb, Sticky Barb, Air Balloon, Absorb Bulb, Cell Battery, Snowball, Luminous Moss, Kee Berry, Maranga Berry

## Test plan
- [x] 141 tests pass covering all item triggers and helpers
- [x] All 281 gen9 tests pass (no regressions)
- [x] TypeScript typecheck clean
- [x] Biome lint/format clean

Closes: N/A

---
Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Gen 9 held-item system added with full in-battle behavior for Booster Energy, Fairy Feather, Covert Cloak and other items, covering before-move, on-hit, on-contact, on-damage-taken, and end-of-turn interactions.
  * Centralized Gen 9 held-item handling ensuring consistent activations and suppression under effects like Klutz, Embargo, and Magic Room.

* **Tests**
  * Added 141 tests validating item triggers, consumptions, heals/damage, stat changes, and edge cases.

### CODE REVIEW

File: packages/gen9/src/Gen9Items.ts [hunk 1]
```diff
@@ -0,0 +1,1739 @@
...
+      const threshold = getPinchBerryThreshold(pokemon, 0.25);
+      if (currentHp > 0 && currentHp <= Math.floor(maxHp * threshold)) {
+        return {
+          activated: true,
+          effects: [
+            { type: "stat-boost", target: "self", value: "critRatio" },
+            { type: "consume", target: "self", value: "lansat-berry" },
+          ],
+          messages: [`${pokemonName}'s Lansat Berry raised its critical-hit ratio!`],
+        };
```
[the remark this answers is not in the document]
Comment (uehlbran) [reply packages/gen9/src/Gen9Items.ts]: Confirmed — `stat-boost critRatio` is not recognized by the engine. The proper fix requires a new `lansat-crit` volatile status and an override of `rollCritical()` in `Gen9Ruleset`. Filed as bug #726 for tracking. The current behavior (berry consumed, no crit boost) will be fixed in a dedicated follow-up.
